README.md

```diff
@@ -21,4 +21,112 @@ An acme client that integrates with Kubernetes to automatically generate and ren
+  acme-reg.json: ewogICJ0ZXJtc19vZl9zZXJ2aWNlIjogImh0dHBzOi8vbGV0c2VuY3J5cHQub3JnL2RvY3VtZW50cy9MRS1TQS12MS4wLjEtSnVseS0yNy0yMDE1LnBkZiIsCiAgIm5ld19hdXRoenJfdXJpIjogImh0dHBzOi8vYWNtZS1zdGFnaW5nLmFwaS5sZXRzZW5jcnlwdC5vcmcvYWNtZS9uZXctYXV0aHoiLAogICJ1cmkiOiAiaHR0cHM6Ly9hY21lLXN0YWdpbmcuYXBpLmxldHNlbmNyeXB0Lm9yZy9hY21lL3JlZy8xNzg5NzkiLAogICJib2R5IjogewogICAgImFncmVlbWVudCI6ICJodHRwczovL2xldHNlbmNyeXB0Lm9yZy9kb2N1bWVudHMvTEUtU0EtdjEuMC4xLUp1bHktMjctMjAxNS5wZGYiLAogICAgImNvbnRhY3QiOiBbCiAgICAgICJtYWlsdG86ZGVtb0BkZW1vLmNvbSIKICAgIF0sCiAgICAia2V5IjogewogICAgICAieSI6ICJLYmZPY3c1X0NHVWFVOXoxS1ljWVo2ODItbTV0OXYyT05NUTFIVV80dGZTYlF0SEE5cTQ0T0NpLS0xbDdVX291IiwKICAgICAgIngiOiAiZFR4bGFPdjNuUDZSaXY5Qk1ES0U4SmFVaEpyUDB4NTFHWjBITmo4VnZwdEM4WGdVTElGMFBBVlptWEJZUGlrUCIsCiAgICAgICJjcnYiOiAiUC0zODQiLAogICAgICAia3R5IjogIkVDIgogICAgfSwKICAgICJpZCI6IDE3ODk3OSwKICAgICJyZXNvdXJjZSI6ICJyZWciCiAgfQp9Cg==
+```
+
+To generate this, you can use `example/getusersecret.sh -e your@email.com -s acme-staging.api.letsencrypt.org`. 
+This one obviously uses staging letsencrypt servers so you need to change to production one (acme-v01.api.letsencrypt.org) when going prod.
+
+At this point you can create a service and deployment in your namespace from `example/deployment.yaml` and `example/service.yaml`. 
+Remember to update --acme-email and --acme-server to their correct values in the deployment.
+
+You now have kube-acme ready to handle your certificates in the namespace they are provisioned in.
+
+### Setting ingress to use ACME secrets
+
+Assuming you have completed the initial setup described above, you can now proceed with defining acme enabled ingress. 
+Start by creating a new secret that your ingress will reference, and make sure the secret is acme enabled
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: some.domain.tld-acmetls
+  labels:
+    acme-managed: 'true'
+type: Opaque
+data:
+```
+
+> Remember this should be done once, and once only. If you apply this secret again you may end up wiping a valid certificate.
+
+Before you proceed with creating ingresses to handle your incoming traffic it is preferable that you alredy point 
+the domain name to the ingress controller IP address. In some cases (ie. when each ingress gets a separate IP 
+after it is created) this will not be possible. In such situations you need to create the ingress, point the domain name 
+to the new assigned IP and then modify ingress (ie. by adding/changing a custom label) so that `kube-acme` is forced 
+to attempt certificate generation again. Before you create the actual TLS enabled ingress, you need an HTTP ingress 
```

Comment:
"Before you create the actual TLS enabled ingress, you need an HTTP ingress to respond to acme-challenge."

feel this could be clearer.

```
As kube-acme must respond to challenge requests via HTTP (not HTTPS), your ingress controller must route unencrypted traffic for the `/.well-known/acme-challenge` to kube-acme. At the moment, how this is best to be achieved is dependant on your ingress controllers implementation. A few options are as follows:

...
```

There's some discussion about how best to achieve configuration of the endpoint here: https://github.com/munnerz/kube-acme/issues/9 that you may want to use as a reference